src/templates/apiReference.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useMemo } from 'react';
+import { Helmet } from 'react-helmet';
+import PropTypes from 'prop-types';
+import { graphql, navigate, Script, ScriptStrategy } from 'gatsby';
+import Layout from '../components/Layout';
+import Html from '../components/blocks/Html';
+import { LeftSideBar } from '../components/StaticQuerySidebar';
+import PageLanguageContext, { PageLanguagesContext } from '../contexts/page-language-context';
+import Article from '../components/Article';
+import { DEFAULT_LANGUAGE, IGNORED_LANGUAGES } from '../../data/createPages/constants';
+import RightSidebar from '../components/Sidebar/RightSidebar';
+import PageTitle from '../components/PageTitle';
+import { DOCUMENTATION_PATH } from '../../data/transform/constants';
+import { safeWindow } from '../utilities/browser/safe-window';
+import { PREFERRED_LANGUAGE_KEY } from '../utilities/language/constants';
+import { createLanguageHrefFromDefaults, getLanguageDefaults } from '../components/common/language-defaults';
+
+const getMetaDataDetails = (document, prop, alternative = '') =>
+  document && document.meta && document.meta[prop] ? document.meta[prop] : alternative;
+
+const CANONICAL_ROOT = `https://www.ably.com${DOCUMENTATION_PATH}`;
+const META_DESCRIPTION_FALLBACK = `Ably provides a suite of APIs to build, extend, and deliver powerful digital experiences in realtime. Organizations like Toyota, Bloomberg, HubSpot, and Hopin depend on Ably’s platform to offload the growing complexity of business-critical realtime data synchronization at global scale.`;
+const NO_LANGUAGE = 'none';
+
+const ApiReference = ({
+  location: { search },
+  pageContext: { contentOrderedList, languages, version, contentMenu, slug, script },
+  data: { document, versions },
+}) => {
+  const params = new URLSearchParams(search);
+  const language = params.get('lang') ?? DEFAULT_LANGUAGE;
+  useEffect(() => {
+    const preferredLanguage = safeWindow.localStorage.getItem(PREFERRED_LANGUAGE_KEY);
```

Comment:
I've looked at how `safeWindow` is implemented, I think it could use some changes. We shouldn't conflate accessing the window with accessing localStorage - those are 2, very different concerns.

Instead, this should be something like:
```
... = storage.getItem(PREFERRED_LANGUAGE_KEY)
```

And this figures out the window, and returns nothing if none is set or we're not in a browser environment.